src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class MvcEndpointRouteBuilderExtensions
+    {
+        public static IEndpointConventionBuilder MapMvcControllers(
+            this IEndpointRouteBuilder routeBuilder)
+        {
+            return MapMvcControllers<ControllerBase>(routeBuilder);
+        }
+
+        public static IEndpointConventionBuilder MapMvcControllers<TController>(
+            this IEndpointRouteBuilder routeBuilder) where TController : ControllerBase
```

Comment:
This is a neat idea. Controllers don't actually require inheriting controllerbase though.